File: common/src/main/java/com/hartwig/actin/molecular/datamodel/evidence/ExternalTrial.java
package com.hartwig.actin.molecular.datamodel.evidence;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class ExternalTrial {

    @NotNull
    public abstract String title();

    @NotNull
    public abstract Set<String> countries();

    @NotNull
    public abstract String url();

    @NotNull
    public abstract String nctId();
}

File: datamodel/src/main/kotlin/com/hartwig/actin/datamodel/molecular/evidence/ExternalTrial.kt
package com.hartwig.actin.datamodel.molecular.evidence

import com.hartwig.actin.datamodel.trial.TrialPhase

data class ExternalTrial(
    val nctId: String,
    val title: String,
    val acronym: String?,
    val phase: TrialPhase?,
    val genderMatch: Boolean?,
    val treatments: Set<String>,
    val countries: Set<CountryDetails>,
    val molecularMatches: Set<MolecularMatchDetails>,
    val applicableCancerTypes: Set<CancerType>,
    val url: String
) : Comparable<ExternalTrial> {

    override fun compareTo(other: ExternalTrial): Int {
        return title().compareTo(other.title())
    }

    fun title() : String {
        return acronym ?: title
    }
}